From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 29 Oct 2026 16:18:51 +0100
Subject: [PATCH] PmsLogging: PmsLogNonClass() must not format the message
 twice

PmsLogNonClass(), behind the MSG_*2() macros, formats the caller's
message into a buffer and then passes that buffer as the format string of
PmsErrorLog(), PmsNoticeLog() and the others, with a NULL argument. Any
'%' in the message is interpreted a second time and reads arguments that
were never passed. -Wformat-security does not catch it because the call
has an argument after the format.

Pass the buffer through "%s".

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 include/public/pmscore/PmsLogging.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/public/pmscore/PmsLogging.h b/include/public/pmscore/PmsLogging.h
index 3074e9a..d754c00 100644
--- a/include/public/pmscore/PmsLogging.h
+++ b/include/public/pmscore/PmsLogging.h
@@ -156,15 +156,15 @@ public:
         va_end(ar);
         switch(level)
         {
-            case kPmLogLevel_Error   : PmsErrorLog(pLogger, err, DUP(err),"", PMS_FUNCTION,buffer, NULL);
+            case kPmLogLevel_Error   : PmsErrorLog(pLogger, err, DUP(err),"", PMS_FUNCTION, "%s", buffer);
                                        break;
-            case kPmLogLevel_Notice  : PmsNoticeLog(pLogger, "", PMS_FUNCTION,buffer, NULL);
+            case kPmLogLevel_Notice  : PmsNoticeLog(pLogger, "", PMS_FUNCTION, "%s", buffer);
                                        break;
-            case kPmLogLevel_Info    : PmsInfoLog(pLogger, "", PMS_FUNCTION,buffer, NULL);
+            case kPmLogLevel_Info    : PmsInfoLog(pLogger, "", PMS_FUNCTION, "%s", buffer);
                                        break;
-            case kPmLogLevel_Warning : PmsWarningLog(pLogger, "", PMS_FUNCTION,buffer, NULL);
+            case kPmLogLevel_Warning : PmsWarningLog(pLogger, "", PMS_FUNCTION, "%s", buffer);
                                        break;
-            case kPmLogLevel_Debug   : PmsDebugLog(pLogger, "", PMS_FUNCTION, buffer, NULL);
+            case kPmLogLevel_Debug   : PmsDebugLog(pLogger, "", PMS_FUNCTION, "%s", buffer);
                                        break;
             default:                   break;
         }
